test/units/modules/cloud/amazon/test_kinesis_stream.py

```diff
@@ -230,9 +236,21 @@ def test_retention_action_invalid(self):
         )
         self.assertFalse(success)
 
+    def test_update_shard_count(self):
+        client = boto3.client('kinesis', region_name=aws_region)
```

Comment:
You shouldn't be creating a boto3 client directly; use the boto3_conn() function as it is done above so that it's all consistent.  In some cases ansible logic for finding the connection information may be different from standard AWS logic (e.g. where connection information parameters are passed direct to the module) so doing this might lead to inconsistency.